src/schemas.jl

```diff
@@ -661,8 +661,8 @@ function _generate_record_type_definitions(schema_version::SchemaVersion, record
     end
 
     # generate `base_overload_definitions`
-    equal_rhs_statement = foldr((x, y) -> :($x && $y), (:(a.$f == b.$f) for f in keys(record_fields)))
-    isequal_rhs_statement = foldr((x, y) -> :($x && $y), (:(isequal(a.$f, b.$f)) for f in keys(record_fields)))
+    equal_rhs_statement = foldr((x, y) -> :($x && $y), (:(a.$f == b.$f) for f in keys(record_fields)); init=true)
+    isequal_rhs_statement = foldr((x, y) -> :($x && $y), (:(isequal(a.$f, b.$f)) for f in keys(record_fields)); init=true)
```

Comment:
Note that the case that this change is intended to fix is already handled by the approach in #103.